SCClassLibrary/JITLib/Patterns/Pdef.sc

```diff
@@ -345,16 +345,18 @@ TaskProxy : PatternProxy {
 	////////// playing interface //////////
 
 	playOnce { arg argClock, doReset = (false), quant;
-		var clock = argClock ? this.clock;
-		^PauseStream.new(this.asProtected.asStream).play(clock, doReset, quant ? this.quant)
+		var pauseStream = PauseStream(
```

Comment:
This doesn't fix `Tdef().asStream`, does it? Why should independent streams be handled properly if the user calls `playOnce` (which few users would guess) but not `.asStream.play` (which a lot of users would try, I'd imagine)?

Reply:
Oh... there's a really easy answer to my question: because the player is not available in `.asStream`. Because the error handler updates the player's state, it's literally impossible to establish the error handler in a place that is not also establishing the player.

So the rule (which might be worth documenting) is: if the Tdef object makes the player, it will update the player's state on error. If you independently make a stream from a Tdef and play it by yourself, then you're responsible for the player's state.

So this is fine.